lua/lspconfig/server_configurations/rubocop.lua

```diff
@@ -2,10 +2,17 @@ local util = require 'lspconfig.util'
 
 return {
   default_config = {
-    cmd = { 'bundle', 'exec', 'rubocop', '--lsp' },
+    cmd = { 'rubocop', '--lsp' },
     filetypes = { 'ruby' },
     root_dir = util.root_pattern('Gemfile', '.git'),
   },
+  on_new_config = function(config, root_dir)
+    -- prepend 'bundle exec' to cmd if bundler is being used
+    local gemfile_path = util.path.join(root_dir, 'Gemfile')
+    if util.path.exists(gemfile_path) then
+      config.cmd = { 'bundle', 'exec', unpack(config.cmd) }
```

Comment:
@gongfarmer @glepnir FYI, I believe this change has broken my use-case, which was working previously. I am now unable to override `cmd`. 

I am in a rails app, so bundle is available. But the version of rubocop installed in my rails app does not have the lsp. I would like to use the globally installed version, which I was previously doing by overriding cmd:
```
  cmd = { "rubocop", "--lsp", },
```

But now this seems to not be possible. I have also tried overriding `on_new_config`, but my overridden function doesn't seem to get called.

```
lspconfig.rubocop.setup({
  cmd = { "rubocop", "--lsp", },
  on_attach = on_attach,
  capabilities = capabilities,
  on_new_config = function (new_config, new_root_dir)
    print("am i getting called?")
  end,
})
```

I never see the "am i getting called?" message. Additionally, this shows up in the lsp log, suggesting the default on_new_config is still active.
```
[ERROR][2023-07-08 15:38:29] .../vim/lsp/rpc.lua:734	"rpc"	"bundle"	"stderr"	"invalid option: --lsp\nFor usage information, use --help\n"
```

Reply:
Update: I think I've figured this out. Providing my own `on_new_config` implementation doesn't override the default. Instead, both are executed. So I need to actually overwrite the effects of the default `on_new_config`.

My working configuration:

```
lspconfig.rubocop.setup({
  on_attach = on_attach,
  capabilities = capabilities,
  on_new_config = function (new_config, new_root_dir)
    new_config.cmd = { "rubocop", "--lsp", }
  end,
})
```